models/project/board.go

```diff
@@ -244,27 +258,46 @@ func (p *Project) GetBoards(ctx context.Context) (BoardList, error) {
 	return append([]*Board{defaultB}, boards...), nil
 }
 
-// getDefaultBoard return default board and create a dummy if none exist
+// getDefaultBoard return default board and ensure only one exists
 func (p *Project) getDefaultBoard(ctx context.Context) (*Board, error) {
-	var board Board
-	exist, err := db.GetEngine(ctx).Where("project_id=? AND `default`=?", p.ID, true).Get(&board)
-	if err != nil {
+	var boards []Board
+	if err := db.GetEngine(ctx).Where("project_id=? AND `default` = ?", p.ID, true).OrderBy("sorting").Find(&boards); err != nil {
 		return nil, err
 	}
-	if exist {
+
+	// create a default board if none is found
+	if len(boards) == 0 {
```

Comment:
As I said above, I think it is good to make the system self-healing. I strongly against to show a lot of unnecessary 500 errors to end users. Maybe TOC should have a final decision for this.

Reply:
But we don't know what happened. Maybe the last default record can be recovered? But we should not create a new one. I think it's good for self-healing. But we don't know whether creating a new one is right here.

Reply:
> But we don't know what happened.

Even if you know, what could you do?


> Maybe the last default record can be recovered? 

What's the difference? Even if a new board is created, the issues could also be moved to the correct one.

> Maybe the last default record can be recovered? 

I think it is good enough because I can't see a real bad case. Or it could add an error log (actually few people really look at server logs)